src/components/Lesson/Lesson.js

```diff
@@ -40,6 +40,39 @@ const LessonButton = ({path, lessons, t}) => {
   return <InstructionButton buttonPath={buttonPath} buttonText={t('lessons.tolesson')}/>;
 };
 
+const onclickAndSetCheckboxes = (path, setCheckboxes) => {
+  const checkboxes = document.getElementsByTagName('input');
+  const lessonPath = '/' + path;
+  const checkboxProgress = {};
+  for (let i = 0; i < checkboxes.length; i++) {
+    const keyNames = checkboxes[i].id;
+    if (localStorage[lessonPath] === undefined) {
+      setCheckboxes(lessonPath, {});
+    }
+    const checkboxKeys = JSON.parse(localStorage[lessonPath]);
+    if (checkboxes[i].type === 'checkbox') {
+      for (let j = 0; j < localStorage.length; j++) {
+        if (lessonPath === localStorage.key(j) && checkboxKeys[keyNames] === true) {
+          checkboxes[i].setAttribute('checked','true');
+        }
+      }
+    }
+    checkboxProgress[checkboxes[i].id] = false;
+    const myStore = (e) => {
+      if (e.target.checked) {
+        checkboxProgress[checkboxes[i].id] = e.target.checked;
+      }
+      else {
+        checkboxProgress[checkboxes[i].id] = e.target.checked;
+      }
```

Comment:
As mentioned before, this if-block kan be written `checkboxProgress[checkboxes[i].id] = !!e.target.checked`